lib/ansible/modules/cloud/amazon/redshift.py

```diff
@@ -215,14 +227,48 @@
             description: status of the enhanced vpc routing feature.
             returned: success
             type: boolean
+        tags:
+            description: aws tags for cluster.
+            returned: success
+            type: dict
 '''
 
 try:
     import botocore
 except ImportError:
     pass  # handled by AnsibleAWSModule
 from ansible.module_utils.ec2 import ec2_argument_spec, snake_dict_to_camel_dict
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_tag_list, boto3_tag_list_to_ansible_dict
+from ansible.module_utils.ec2 import compare_aws_tags
 from ansible.module_utils.aws.core import AnsibleAWSModule, is_boto3_error_code
+from ansible.module_utils.aws.iam import get_aws_account_id
+
+
+def _ensure_tags(redshift, identifier, existing_tags, module):
+    """Compares and update resource tags"""
+
+    account_id = get_aws_account_id(module)
+    region = module.params.get('region')
+    resource_arn = "arn:aws:redshift:{0}:{1}:cluster:{2}" .format(region, account_id, identifier)
+    tags = module.params.get('tags')
```

Comment:
Does this try to update tags if they are None (the default)? Tags should only be updated if a dictionary is provided.

Besides that, this looks good. You could open a PR against https://github.com/mattclay/aws-terminator/blob/master/aws/policy/big-data.yaml to add the permissions so this can pass in CI.

Reply:
ok added a check for that.

Reply:
@s-hertel I added https://github.com/ansible/ansible/pull/56981/files#diff-fd4728adea42973936ca9bdb0082deadR256
so if the tags array is none it won't make any change. I've mostly copied the logic from https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/aws/rds.py#L214 . 
Should I add a specific test case for this?

Reply:
@rafaeldriutti Yes please on the test case